specs/DisableNavigatingBackAndForward.md

```diff
@@ -0,0 +1,124 @@
+CHECK_FAILURE(m_webView->add_NavigationStarting(
+    Callback<ICoreWebView2NavigationStartingEventHandler>(
+        [this](ICoreWebView2* sender, ICoreWebView2NavigationStartingEventArgs* args)
+        -> HRESULT {
+            wil::com_ptr<ICoreWebView2NavigationStartingEventArgs3> args3;
+            if (SUCCEEDED(args->QueryInterface(IID_PPV_ARGS(&args3))))
+            {
+                COREWEBVIEW2_NAVIGATION_KIND kind;
+                CHECK_FAILURE(args3->get_NavigationKind(&kind));
+                // disable navigation if it is back/forward
+                if (kind == COREWEBVIEW2_NAVIGATION_KIND_BACKORFORWARD)
+                {
+                     CHECK_FAILURE(args->put_Cancel(true));
+                }
+            }
+            return S_OK;
+        })
+    .Get(),
+    &m_navigationStartingToken));
+//! [NavigationStarting]
+```
+
+#### .NET and WinRT
+
+#### Use `CoreWebView2NavigationStartingEventArgs` in `NavigationStarting`
+
+```c#
+// Register a handler for the NavigationStarting event.
+// This handler will check the navigation status, and if the navigation is
+// `GoBack` or `GoForward`, it will be canceled.
+void WebView_NavigationStarting(object sender, CoreWebView2NavigationStartingEventArgs e)
+{
+    if (e.NavigationKind == CoreWebView2NavigationKind.BackOrForward)
+    {
+        e.Cancel = true;
+    }
+}
+```
+
+# API Details
+#### Win32 C++
+
+```c++
+// Enums and structs
+[v1_enum]
+typedef enum COREWEBVIEW2_NAVIGATION_KIND {
+  /// A navigation caused by CoreWebView2.Reload(), location.reload(), the end user using F5 or other UX, or other reload mechanisms to reload the current document without modifying the navigation history.
+  COREWEBVIEW2_NAVIGATION_KIND_RELOAD,
+  /// A navigation back or forward to a different entry in the session navigation history. For example via CoreWebView2.Back(), location.back(), the end user pressing Alt+Left or other UX, or other mechanisms to navigate forward or backward in the current session navigation history.
+  COREWEBVIEW2_NAVIGATION_KIND_BACKORFORWARD,
+  /// A navigation to a different document. This can be caused by CoreWebView2.Navigate(), window.location.href = '...', or other WebView2 or DOM APIs that navigate to a specific URI.
+  COREWEBVIEW2_NAVIGATION_KIND_DIFFERENT,
+} COREWEBVIEW2_NAVIGATION_KIND;
+
+/// Extend `NavigationStartingEventArgs` by adding more information.
+[uuid(39A27807-2365-470B-AF28-885502121049), object, pointer_default(unique)]
+interface ICoreWebView2NavigationStartingEventArgs3 : ICoreWebView2NavigationStartingEventArgs2 {
+
+  /// Indicates if this navigation is reload, back/forward or navigating to a different document 
+  [propget] HRESULT NavigationKind(
+      [out, retval] COREWEBVIEW2_NAVIGATION_KIND* kind);
+}
+}
+```
+
+#### .NET and WinRT
+
+```c# (but really MIDL3)
+namespace Microsoft.Web.WebView2.Core
+{
+    enum CoreWebView2NavigationKind
+    {
+        Reload = 0,
+        BackOrForward = 1,
```

Comment:
There is currently no such demand, and more importanly, we could not distinguish from origin source `NavigationType`. Therefore, if there is such a need later, we can achieve it in other ways.

Reply:
Note that this prevents us from ever adding `COREWEBVIEW2_NAVIGATION_KIND_FORWARD` and `COREWEBVIEW2_NAVIGATION_KIND_BACK` later. From the discussion, it appears that the team accepts this, and if it becomes an issue, they will provide an alternative.

Reply:
That answers my question too. Spec process note, it's helpful to add this as a note in the spec for the record. Not something that will go into the public doc, but when the next person comes along in a year and re-asks the question. Something like:

_Spec note: This doesn't distinguish forward from backward navigations because we can't distinguish it from origin source `NavigationType`._

Reply:
It makes sense. Mark it.